docs/adr/002-application-architecture-for-configurability.md

```diff
@@ -0,0 +1,117 @@
+
+We also need to take into account that the evolution of the application with new and improved features needs to continue and we have limited team resources.
+
+
+## Decision Drivers
+
+- Investment payoff
+- Compatibility with continued support for instances with feature evolution
+- Improve developer velocity
+- Ease of support for required customizations
+
+
+## Considered Options
+
+- [1] Continue with current application architecture + design system
+- [2] Continue with current application architecture + replace design system
+- [3] Refactor component library + replace design system + rewrite instances
+- [4] Rewrite component library + replace design system + rewrite instances
+
+## Decision Outcome
+
+- ✔️[3] Refactor component library + replace design system + rewrite instances
+
+Move more control to the instance level, modularize the core ui library to expose core feature components and smaller reusable components, and create a data layer as part of the core ui library that is ideally also consumed at the instance level that manages data fetching and MDX parsing so that components are data agnostic. Introduce a new design system that is well-maintained and community-backed to create standardization across styles and accessibility patterns.
+
+#### 1. Move Routing from the core ui library to the instance level
+Routing and what pages exist should be determined at the instance level. Currently, routing is handled at the core UI library level so supporting additional/removal of pages requires override logic. Moving this to the instance level allows developers/stakeholders to independently manage their routes and pages.
+
+#### 2. Feature components in the core ui library composed at the instance level
+Currently, the core UI library handles the composition of each page because of its control over routing. However, pages and what they render should shift to the instance level so developers/stakeholders can control what is rendered within each page view. The core UI library should expose feature components (containers) that deliver a core feature in the VEDA dashboard such as the A&E feature. 
+
+#### 3. Modularize and create smaller reusable components
+The core UI library should also expose smaller reusable components like date pickers, form elements, analysis tools, etc… (“presentational/dumb” components). This way developers/stakeholders for example can compose a page view with the A&E feature from the core UI library and consume other reusable components to construct their page view. This also now allows developers to build their own custom components on the instance side and directly incorporate them. 
+
+#### 4. Create a Data layer that manages all data fetching
```

Comment:
I'd like to clarify my understanding to ensure I've grasped this correctly.

This data utility will be consumed at the instance level and will pass the 'reconciled' data to the component (in JSON format, I think). Additionally, the utility will be optional, allowing those with a CMS system different from MDX the flexibility to format their data independently and then pass the finalized JSON to the components. In other words, we will only offer data utility for MDX files with specific frontmatter. 

If my understanding aligns with the intended implementation, could we possibly integrate this explanation into the content?

Reply:
Yes it is. I mainly wanted to make it clear that we are not going to offer various types of contents parser, other than MDX 👍